### DISCUSSION TITLE
How to set LLM enviroment when USE nxadb.Graph.chat()?

### DISCUSSION THREAD

**Discussion by qiangziakbar** [arangodb/nx-arangodb#63] [Q&A]
when use nxadb.Graph.chat("hello"), reports as below:
ModuleNotFoundError: LLM dependencies not installed. Install with **pip install nx-arangodb[llm]**

how to set api-key? how to set LLM model?

**Comment by aMahanna** [comment]
Hi @qiangziakbar 

After installing with `pip install nx-arangodb[llm]`, you can set the API key with:

```python
import os

os.environ["OPENAI_API_KEY"] = "sk-..."
```

And then you can set the LLM Model using any of the `BaseLanguageModel` classes from [langchain](https://api.python.langchain.com/en/latest/language_models/langchain_core.language_models.base.BaseLanguageModel.html):

```python
import nx_arangodb as nxadb

G = nxadb.Graph(...)

from langchain_openai import ChatOpenAI # for example

G.chat("hello", llm=ChatOpenAI(temperature=0, model_name="gpt-4"))
```
